packages/frontend/src/creation-form/components/generic-data/index.tsx

```diff
@@ -82,39 +65,42 @@ export const GenericData = ({
 
     useEffect(() => {
         setDisabled(
-            !title ||
-                !description ||
-                !erc20Name ||
-                !erc20Symbol ||
-                !title.trim() ||
-                title.trim().length > MAX_KPI_TOKEN_TITLE_CHARS ||
-                !stripHtml(description).trim() ||
-                stripHtml(description).trim().length >
+            !state.title ||
+                !state.description ||
+                !state.erc20Name ||
+                !state.erc20Symbol ||
+                !state.title.trim() ||
+                state.title.trim().length > MAX_KPI_TOKEN_TITLE_CHARS ||
+                !stripHtml(state.description).trim() ||
+                stripHtml(state.description).trim().length >
                     MAX_KPI_TOKEN_DESCRIPTION_CHARS ||
-                tags.length === 0 ||
-                tags.length > MAX_KPI_TOKEN_TAGS_COUNT ||
-                !expiration ||
-                isInThePast(expiration) ||
-                !erc20Name.trim() ||
-                erc20Name.trim().length > MAX_KPI_TOKEN_ERC20_NAME_CHARS ||
-                !erc20Symbol.trim() ||
-                erc20Symbol.trim().length > MAX_KPI_TOKEN_ERC20_SYMBOL_CHARS ||
-                !erc20Supply.value ||
-                parseFloat(erc20Supply.value) === 0
+                !state.tags ||
+                state.tags.length === 0 ||
+                state.tags.length > MAX_KPI_TOKEN_TAGS_COUNT ||
+                !state.expiration ||
+                isInThePast(state.expiration) ||
+                !state.erc20Name.trim() ||
+                state.erc20Name.trim().length >
+                    MAX_KPI_TOKEN_ERC20_NAME_CHARS ||
+                !state.erc20Symbol.trim() ||
+                state.erc20Symbol.trim().length >
+                    MAX_KPI_TOKEN_ERC20_SYMBOL_CHARS ||
+                !state.erc20Supply ||
+                !state.erc20Supply.value ||
+                parseFloat(state.erc20Supply.value) === 0
```

Comment:
It would be awesome to improve this code, maybe abstract this logic in functions?

Reply:
Addressed in 6694ad211b2427ae30c7924510a3b135270fdb49